python/lsst/sims/movingObjects/linearObs.py

```diff
@@ -0,0 +1,293 @@
+        """
+        interpfuncs = {}
+        for n in ephs.dtype.names:
+            if n == 'time':
+                continue
+            interpfuncs[n] = interpolate.interp1d(ephs['time'][i], ephs[n][i], kind='linear',
+                                                  assume_sorted=True, copy=False)
+        return interpfuncs
+
+    def _setupCamera(self):
+        """
+        Initialize camera mapper, etc.
+        """
+        self.mapper = LsstSimMapper()
+        self.camera = self.mapper.camera
+        self.epoch = 2000.0
+        self.cameraFov = np.radians(2.1)
+
+    def ssoInFov(self, interpfuncs, simdata, rFov=np.radians(1.75),
+                 useCamera=True,
+                 simdataRaCol = 'fieldRA', simdataDecCol='fieldDec'):
+        """
+        Return the indexes of the simdata observations where the object was inside the fov.
+        """
+        # See if the object is within 'rFov' of the center of the boresight.
+        raSso = np.radians(interpfuncs['ra'](simdata['expMJD']))
+        decSso = np.radians(interpfuncs['dec'](simdata['expMJD']))
+        sep = haversine(raSso, decSso, simdata[simdataRaCol], simdata[simdataDecCol])
+        if not useCamera:
+            idxObsRough = np.where(sep<rFov)[0]
+            return idxObsRough
+        # Or go on and use the camera footprint.
+        try:
+            self.camera
+        except AttributeError:
+            self._setupCamera()
+        idxObs = []
+        idxObsRough = np.where(sep<self.cameraFov)[0]
+        for idx in idxObsRough:
+            mjd_date = simdata[idx]['expMJD']
+            mjd = ModifiedJulianDate(TAI=mjd_date)
+            obs_metadata = ObservationMetaData(pointingRA=np.degrees(simdata[idx][simdataRaCol]),
+                                               pointingDec=np.degrees(simdata[idx][simdataDecCol]),
+                                               rotSkyPos=np.degrees(simdata[idx]['rotSkyPos']),
+                                               mjd=mjd)
+            raObj = np.array([interpfuncs['ra'](simdata[idx]['expMJD'])])
+            decObj = np.array([interpfuncs['dec'](simdata[idx]['expMJD'])])
+            # Catch the warnings from astropy about the time being in the future.
+            with warnings.catch_warnings(record=False):
+                warnings.simplefilter('ignore')
+                chipNames = chipNameFromRaDec(ra=raObj,dec=decObj, epoch=self.epoch,
+                                              camera=self.camera, obs_metadata=obs_metadata)
+            if chipNames != [None]:
+                idxObs.append(idx)
+        idxObs = np.array(idxObs)
+        return idxObs
+
+
+    def _calcColors(self, sedname='C.dat'):
+        """
+        Calculate the colors for a moving object with sed 'sedname'.
+        """
+        # Do we need to read in the LSST bandpasses?
+        try:
+            self.lsst
+        except AttributeError:
+            filterdir = os.getenv('LSST_THROUGHPUTS_BASELINE')
```

Comment:
You're totally right about the power that be saying to use getPackageDir, in case they change how eups works. However, I'm not sure they'd change eups enough that env variables set within the package don't work anymore (I interpreted it as, the env variable set by eups to indicate where the package lives might go away).